controllers/user.js

```diff
@@ -1,35 +1,114 @@
-const User = require('../models/user');
+require('dotenv').config();
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const User = require('../dbfunction/users');
+
+function generateToken(user) { return jwt.sign(user.toJSON(), process.env.ACCESS_TOKEN_SECRET, { expiresIn: '1w' }); }
 
 const UserController = {
   signup: async (req, res) => {
     try {
       const {
         firstname, lastname, email, password,
       } = req.body;
-      const user = await User.create({
-        firstname,
-        lastname,
-        email,
-        password,
-      }, {
-        fields: ['firstname', 'lastname', 'email', 'password'],
-      });
+      // check if user exist
+      const userExist = await User.userExist(email);
+      if (userExist) {
+        return res.status(409).send({
+          success: false,
+          message: `${userExist.email} already exist`,
+        });
+      }
+
+      // check for empty fields
+      if (!firstname || !lastname || !email || !password) {
+        return res.status(400).send({
+          success: false,
+          message: 'complete the empty fields before submitting',
+        });
+      }
```

Comment:
I am thinking you can move this validation as the first validation. the reason is that you don't want to do further processing (like checking the DB) if any of the payload is not sent by the user, hence we return execution as early as possible
cc: @richard